### PR TITLE
Switched to hatch + deleted unneeded setup.cfg and setup.py

### PR DESCRIPTION
**Pull request by sverhoeven**
Also moves mypy from precommit to ci job

Refs #441

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -17,3 +98,16 @@ max-line-length = "88"
...
 [tool.pydocstyle]
 convention = "google"
+
+[tool.coverage.run]
+source= "src"
+branch = true
+parallel = true
+
+[tool.pytest.ini_options]
+testpaths = ["tests"]
```
Comment (BSchilperoort) [inline pyproject.toml:115]: src is missing here (for doctest)

File: pyproject.toml [hunk 2]
```diff
@@ -17,3 +98,16 @@ max-line-length = "88"
...
+
+[tool.mypy]
+plugins = ["pydantic.mypy"]
+ignore_missing_imports = true
+files = ["src", "tests"]
```
Comment (BSchilperoort) [inline pyproject.toml]: I'm not a big fan of checking the typing of the tests, can be a hassle.

Also, we can add `check_untyped_defs = True` (but that might give some new mypy errors)

`warn_unused_ignores = True` would be good to add.

**Review by BSchilperoort — APPROVED** [review]
Looks good, only some small comments.

The built tarball includes many files currently, and MANIFEST.in is being ignored (can be deleted). You can use excludes to reduce the tarball size.

File: pyproject.toml (hunk 1 above)
Comment (sverhoeven) [reply pyproject.toml:115]: Skipped as most docstrings have unmet expectations, added as nice to have to #440

File: pyproject.toml (hunk 2 above)
Comment (sverhoeven) [reply pyproject.toml]: Added both

**Comment by sverhoeven** [comment]
Thanks for checking and your suggestions.

**Comment by BSchilperoort** [comment]
> Thanks for checking and your suggestions.

No problem.

By the way, MANIFEST.in is still there, but is no longer doing anything

**Comment by sverhoeven** [comment]
> > Thanks for checking and your suggestions.
> 
> No problem.
> 
> By the way, MANIFEST.in is still there, but is no longer doing anything

Made https://github.com/eWaterCycle/ewatercycle/pull/444 to remove it.

**Pull request merged**